### PR TITLE
Add CoreBot to VSIX template

### PR DESCRIPTION
**Pull request by GasparAcevedoZainSouthworks**
## Proposed Changes
  - Add `CoreBot` sample to VSIX templates
  - Update `Readme.md` file with info related to `CoreBot`
  - Also, fixes issue #1073
## Testing
1.  Open the `BotBuilderTemplatesVSIX.sln` in Visual Studio 2017
2.  Once the project is loaded, rebuild it. Once the re-build process is concluded, close the Visual Studio instance.
3.  Go to `{your project path}\bin\debug` folder and run the `BotBuilderVSIXV4.vsix` VSIX installer. Ensure that there isn't any Visual Studio instance open.
4. Once the installation process is concluded, open a Visual Studio 2017 instance and go to `File -> new -> project` and you will see the templates under the `Visual C# -> Cloud -> AzureBotService` category

image

5. To test it in Visual Studio 2019, open a new instance and select  `Create New Project`. You will see the templates in the template selection window.

image

### CODE REVIEW

File: generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.cs [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.Dialogs;
+using Microsoft.Bot.Schema;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace Microsoft.BotBuilderSamples
```
Comment (cleemullins) [inline generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.cs]: This namespace need to be templatized acrosss all the .cs files. For example:
namespace $safeprojectname$

You can see an example here: https://github.com/Microsoft/BotBuilder-Samples/blob/master/generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/EchoBot/CounterState.cs

Comment (GasparAcevedoZainSouthworks) [reply generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.cs]: Feedback applied

**Comment by cleemullins** [comment]
Testing this now.

**Comment by GasparAcevedoZainSouthworks** [comment]
> @GasparAcevedoZainSouthworks Any way to clean up the simple stuff so we can figure out how to merge this in?

@cleemullins we fixed the first three issues. Regarding the fourth issue, we have the same error as you:
image

If you agree, we will create an Issue in this repository, but we consider that this isn't necessary due to the error message is pretty explanative.

File: generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.cs [hunk 2]
```diff
@@ -0,0 +1,271 @@
...
+        {
+            var response = activity.CreateReply();
+            response.Attachments = new List<Attachment>() { attachment };
+            return response;
+        }
+
+        // Load attachment from file.
+        private Attachment CreateAdaptiveCardAttachment()
+        {
+            var adaptiveCard = File.ReadAllText(@".\Dialogs\Welcome\Resources\welcomeCard.json");
```
Comment (sgellock) [inline generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.cs]: This code needs to change to use Path.Combine which is environment agnostic.  I just fixed this in the samples/csharp_dotnet/13.basic-bot sample.  Please use: 
            string[] paths = {".", "Dialogs", "Welcome", "Resources", "welcomeCard.json"};
            string fullPath = Path.Combine(paths);
            var adaptiveCard = File.ReadAllText(fullPath);

File: generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.csproj [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    <TargetFramework>netcoreapp2.1</TargetFramework>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.AspNetCore" Version="2.1.3" />
+    <PackageReference Include="Microsoft.AspNetCore.All" Version="2.0.8" />
+    <PackageReference Include="Microsoft.ApplicationInsights.AspNetCore" Version="2.1.1" />
+    <PackageReference Include="Microsoft.Azure.CognitiveServices.ContentModerator" Version="0.12.1-preview" />
+    <PackageReference Include="Microsoft.Azure.CognitiveServices.Language" Version="1.0.1-preview" />
+    <PackageReference Include="Microsoft.Bot.Builder" Version="4.2.0" />
```
Comment (sgellock) [inline generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.csproj]: I'm not sure when we're going to merge this PR, but we're close enough now that if we decide to wait until 4.3 is released, then all of the Microsoft.Bot.* assembly references should use Version="4.3.0"

File: generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.cs (hunk 2 above)
Comment (GasparAcevedoZainSouthworks) [reply generators/vsix-vs-win/BotBuilderVSIX-V4/UncompressedProjectTemplates/CoreBot/CoreBot.cs]: Feedback Applied

**Pull request merged**